### ISSUE TITLE
Improve generators for numbers

### ISSUE THREAD

**Issue by mgold** [NoRedInk/elm-check#6]
The `int` and `float` generators currently make very rudimentary attempts to produce values that will detect bugs. We are not the first to tackle this problem.
- [QuickCheck arbitrary instances](http://hackage.haskell.org/package/QuickCheck-2.8.2/docs/src/Test-QuickCheck-Arbitrary.html#line-428)
- Figure 4 in [this pager](http://www.ccs.neu.edu/home/stamourv/papers/random-testing-base-type-environment.pdf)

Alternatively, add these as additional generators. For example, I can see "multiplication and division are inverses" failing when the values get big enough to overflow, or being infinite or NaN.

**Comment by mgold** [comment]
One of the oddities of floating point is negative zero. Elm will parse `-0` as positive zero but it can still be obtained:

```
>  1 / 0 > 0
True : Bool
>  1 / -0 > 0
True : Bool
>  1 / (-1 * 0) > 0
False : Bool
```

**Comment by mgold** [comment]
Here's a winkle: NaN doesn't equal itself. So any list of floats that has NaN won't equal itself, causing unrelated tests to fail. So the default `float` generator should not produce NaN.

**Comment by mgold** [comment]
I'm going to call this partially resolved by f652316, and partially awaiting example. Is there a claim that fails because `float` doesn't produce sufficiently varied output? Until then, closing.
